doc/source/ray-air/examples/convert_existing_tf_code_to_ray_air.ipynb

```diff
@@ -0,0 +1,1070 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "6462598d",
+   "metadata": {},
+   "source": [
+    "# Convert existing Tensorflow/Keras code to Ray AIR\n",
+    "\n",
+    "If you already have working Tensorflow code, you don’t have to start from scratch to utilize the benefits of Ray AIR. Instead, you can continue to use your existing code and incrementally add Ray AIR components as needed.\n",
+    "\n",
+    "Some of the benefits you’ll get by using Ray AIR with your existing Tensorflow training code:\n",
+    "\n",
+    "- Easy distributed data-parallel training on a cluster\n",
+    "- Automatic checkpointing/fault tolerance and result tracking\n",
+    "- Parallel data preprocessing\n",
+    "- Seamless integration with hyperparameter tuning\n",
+    "- Scalable batch prediction\n",
+    "- Scalable model serving\n",
+    "\n",
+    "This tutorial will show you how to start with Ray AIR from your existing Tensorflow training code. We will learn how to perform **distributed data-parallel training** and do **scalable batch prediction**."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "69faab49",
+   "metadata": {},
+   "source": [
+    "## Example Code\n",
+    "\n",
+    "The example code we’ll be using is that of the [Tensorflow quickstart tutorial](https://www.tensorflow.org/tutorials/quickstart/beginner). This code trains a neural network classifier on the MNIST dataset.\n",
+    "\n",
+    "Follow along this example by launching the notebook using the 🚀 icon above!"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d6d81045",
+   "metadata": {},
+   "source": [
+    "## Existing Tensorflow Code\n",
+    "\n",
+    "Let’s start with the unmodified code from the example. A thorough explanation of the parts is given in the full tutorial - we’ll just focus on the code here."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "164fc2bc",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "TensorFlow version: 2.9.2\n"
+     ]
+    }
+   ],
+   "source": [
+    "import tensorflow as tf\n",
+    "print(\"TensorFlow version:\", tf.__version__)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7ff0207a",
+   "metadata": {},
+   "source": [
+    "First, we load and preprocess the MNIST dataset.\n",
+    "\n",
+    "Assumption for this tutorial: your existing code is using the `tf.data.Dataset` native to Tensorflow. This tutorial continues to use `tf.data.Dataset` to allow you to make as few code changes as possible. **Everything in this tutorial is also possible if you choose to use Ray Data, and you will also get the benefits of efficient preprocessing and multi-worker batch prediction.** See [here](train-datasets) for resources to get started with Ray Data."
```

Comment:
oh nice does the `See [here](train-datasets)` link actually work?

Reply:
Yeah, all the refs work if you try building the docs from this branch.